src/staticanalysis/bot/static_analysis_bot/revisions.py

```diff
@@ -230,11 +232,33 @@ def __str__(self):
     def url(self):
         return 'https://{}/D{}'.format(self.api.hostname, self.id)
 
+    def setup_try(self, tasks):
+        '''
+        Find the mercurial revision from the Try decision task env
+        '''
+        # Find the decision task
+        def is_decision_task(task):
+            image = task['task']['payload'].get('image')
+            if image is not None and isinstance(image, str):
```

Comment:
Could it not be a string?

Reply:
Yeap, it can be a dictionnary when using [an in-tree docker image](https://tools.taskcluster.net/groups/IWsxR_zMT7O2Us_gjL35gw/tasks/Aos20VUATjKFOLl0ofy3Gw/details), like:

```json
  "image": {
    "path": "public/image.tar.zst",
    "taskId": "dge9dvsMRBOiZEPIyO-aXQ",
    "type": "task-image"
  },
```

https://tools.taskcluster.net/groups/IWsxR_zMT7O2Us_gjL35gw/tasks/Aos20VUATjKFOLl0ofy3Gw/details